gen-view.py

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import re
+from typing import List
+import yaml
+
+CHAPTERS = ["Hardware Software Interface", "Intro Computer Architecture", "Memory Layout", "Memory Security"]
+LABS_PATH = "chapters/"
+
+viewDir = ".view"
+readingDir = f"{viewDir}/reading"
+tasksDir = f"{viewDir}/tasks"
+guidesDir = f"{viewDir}/guides"
+mediaDir = f"{viewDir}/media"
+
+
+
+def hypenate(text: str) -> str:
+    return "-".join(text.strip().split(" ")).lower()
+
+
+def group_reading():
+    """
+    Group all the reading files in a single directory.
+    """
+    os.makedirs(readingDir, exist_ok=True)
+
+    for root, _, files in os.walk(LABS_PATH):
+        for f in files:
+            if "reading" in root and f.endswith(".md") and f != "overview.md":
+                subfolder_name = os.path.basename(os.path.dirname(root))  
+                new_filename = f"{subfolder_name}_reading.md"
+                destination = os.path.join(readingDir, new_filename)
+                os.popen(f"cp {os.path.join(root, f)} {destination}")
```

Comment:
```suggestion
                    taskName = os.path.basename(root)
                    os.popen(f"cp {os.path.join(root, f)} {tasksDir}/{taskName}.md")
```
Why change this from the original script? [1] Or perhaps you were using an older version?

[1] https://github.com/cs-pub-ro/operating-systems/blob/main/gen-view.py

Reply:
I had to change it because all the files would've been named reading.md otherwise. Then the .view/reading folder would've only contained the last reading.md file that was processed. The README.md files in the guides and tasks folders are placed in subfolders, each of them having a different name. Basically, in order to apply the same naming logic to the files in reading, it's the subfolder's name that has to be used (the second-to-last component in the path), not the basename.

Reply:
Ok, now it makes sense. Leave it as it is then.

Reply:
Partially, yes. And I changed the format. However, the links in the other .md files (mostly tasks) still look like this 
![image](https://github.com/user-attachments/assets/c3c7e5b2-8829-4297-a785-bef3ffc134f7)
or like this 
![image](https://github.com/user-attachments/assets/be5efab1-17b6-4c89-8975-42e1d2ed1e96)
Is there a way of solving these links without manually modifying the .md files?

Reply:
Yup. In the `solve_links()` function, you'll just need to change links to `reading/README.md#anchor` to `lab<current>.md#anchor`.

Reply:
In the case of reading/README.md links, without any anchors, there is a problem. There can be multiple files from the reading folder in one lab, so how would we know which one to link to?

Reply:
Then do it in a simpler way: rename all `reading/READING.md`s to `reading/<subchapter-name>.md`. Won't that solve all the issues?

Reply:
I already did in the .view/reading folder

Reply:
I meant to change the file names in the repo and simplify `gen-view.py`.

Reply:
Oh, I got it! Thank you!